### DISCUSSION TITLE
Recoverability: Do task resume after VM shutdown?

### DISCUSSION THREAD

**Discussion by trung** [kagkarlsson/db-scheduler#424] [Q&A]
Hi,

My main use case is to run onetime task but it could be long-running one. 
The question that I have is if the VM shutdown (either gracefully or not). Would the incomplete onetime task be resumed to execution? 

Regards,

Trung

**Comment by kagkarlsson** [comment]
The best way is to gracefully stop the task and resume it again later. There is a similar example here: https://github.com/kagkarlsson/db-scheduler/blob/master/examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/config/LongRunningJobConfiguration.java#L89

Need to regularly check `executionContext.getSchedulerState().isShuttingDown()` in the loop, and return a CompletionHandler that persists the progress and reschedules to a later time (might be Instant.now() for immediate pick by another scheduler)

If not done gracefully, i.e. the `CompletionHandler` is never called, the task will eventually be detected as "dead" and be handled according to its `DeadExecutionHandler` (which by defaults reschedules to now() I think)
